codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/ServerRustSettings.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rust.codegen.server.smithy
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.node.ObjectNode
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.rust.codegen.smithy.CODEGEN_SETTINGS
+import software.amazon.smithy.rust.codegen.smithy.CodegenConfig
+import software.amazon.smithy.rust.codegen.smithy.RuntimeConfig
+import software.amazon.smithy.rust.codegen.smithy.RustSettings
+import java.util.Optional
+
+/**
+ * Configuration of codegen settings
+ *
+ * [renameExceptions]: Rename `Exception` to `Error` in the generated SDK
+ * [includeFluentClient]: Generate a `client` module in the generated SDK (currently the AWS SDK sets this to false
+ *   and generates its own client)
+ *
+ * [addMessageToErrors]: Adds a `message` field automatically to all error shapes
+ * [formatTimeoutSeconds]: Timeout for running cargo fmt at the end of code generation
+ */
+data class ServerCodegenConfig(
+    val renameExceptions: Boolean = true,
+    val includeFluentClient: Boolean = false,
+    val addMessageToErrors: Boolean = true,
+    val formatTimeoutSeconds: Int = 20,
+    // TODO(EventStream): [CLEANUP] Remove this property when turning on Event Stream for all services
+    val eventStreamAllowList: Set<String> = emptySet(),
+) {
+    companion object {
+        fun fromNode(node: Optional<ObjectNode>): CodegenConfig {
```

Comment:
I find it very weird that a method called `fromNode()` on an object does not return itself. Same goes for `from()` on `ServerRustSettings`.

Can we make our settings "inherit" from the client ones and use our settings in the server codegen everywhere? I suspect that in the future we will want to add settings for us that won't be in the client.

Reply:
If not, I think a better design is to have server settings be projected into the client ones where needed in the codebase.

Reply:
Initially I wanted to go for the inheritance as you said. Since we are not changing anything in the client I went for this middle ground approach. Let's hear the @awslabs/rust-sdk-owners thoughts on this and I can refactor.

Reply:
No matter what we still need to fill all the values inside CodegenConfig since we use them in the generation as we share part of the code with the client codegen.